docs/plugins.md

```diff
@@ -0,0 +1,58 @@
+# Plugins
+
+Plugins allow for supporting new transactions without modifying/adding to the core parts of the system. 
+Supporting any new transaction requires:
+-   Defining new transaction types
+-   Defining new message validation rules or updates to any message validation rules
+-   Logic to handle new transactions, validation, making changes to state, etc (for more detail look at the doc on [request handling](request_handling.md)).
+-   Storage of transactions in existing ledger or introducing new ledgers
+-   Changes to existing state or introducing new states  
+-   Defining new constants
+-   Defining any new config variables
+
+
+### Conventions that need to be followed when defining a new plugin:
+#   Each plugin is a separate python package that lives under the package defined in `PLUGIN_ROOT` in `config.py`. It is currently set to `plenum.server.plugin`
```

Comment:
I think its is not necessary to use titles for each rule. Ordered list or bullet points should be enough.
https://github.com/hyperledger/indy-plenum/blob/8d8e34c0d5dfcb1b34d6785e2e826c5274752794/docs/plugins.md